src/scaler/utility/queues/async_priority_queue.py

```diff
@@ -1,70 +1,67 @@
-import heapq
-import sys
 from asyncio import Queue
 from typing import Any, Dict, List, Tuple, Union
 
+from sortedcontainers import SortedDict
+
 PriorityType = Union[int, Tuple["PriorityType", ...]]
 
 
 class AsyncPriorityQueue(Queue):
     """A subclass of Queue; retrieves entries in priority order (lowest first).
 
-    Entries are typically list of the form: [priority, data].
+    Input entries are typically list of the form: [priority, data].
     """
 
     def __len__(self):
         return len(self._queue)
 
     def _init(self, maxsize):
-        self._queue: List[List] = []
+        # self._locator contains {data : List[[priority, data], count]}
+        # self._queue contains {(priority, count) : data}
         self._locator: Dict[bytes, List] = {}
+        self._queue = SortedDict()
```

Comment:
You can add more complete typing:

```suggestion
        self._locator: Dict[bytes, List[Tuple[...]]] = {}
        self._queue: Dict[Tuple[...],  bytes] = SortedDict()
```